### PR TITLE
Release with pgxs

### PR DESCRIPTION
**Pull request by v0idpwn**

### CODE REVIEW

File: images/pgmq-pg/Dockerfile.in [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    && apt-get install -y \
+	ca-certificates \
+	clang \
+	curl \
+	gcc \
+	git \
+	libssl-dev \
+	make \
+	pkg-config \
+	postgresql-server-dev-@@PG_VERSION@@
```
Comment (v0idpwn) [inline images/pgmq-pg/Dockerfile.in:13]: Do we really need both clang and gcc?

File: integration_test/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1 @@
+// Nothing here...
```
Comment (ChuckHend) [inline integration_test/src/lib.rs:1]: Is this file required?

Comment (v0idpwn) [reply integration_test/src/lib.rs:1]: If I remove it, I get:
```
error: failed to parse manifest at `/home/v0idpwn/oss/pgmq/integration_test/Cargo.toml`
Caused by:
  can't find library `pgmq-integration-tests`, rename file to `src/lib.rs` or specify lib.path
```
Maybe there's something we can write on Cargo.toml to avoid this error?

File: Makefile [hunk 3]
```diff
@@ -1,27 +1,33 @@
...
 	docker run -d --name pgmq-pg -e POSTGRES_PASSWORD=postgres -p 5432:5432 quay.io/tembo/pgmq-pg:latest
 
-META.json: META.json.in Cargo.toml
-	@sed "s/@CARGO_VERSION@/$(DISTVERSION)/g" $< > $@
-
-$(DISTNAME)-$(DISTVERSION).zip: META.json
-	git archive --format zip --prefix $(DISTNAME)-$(DISTVERSION)/ --add-file $< -o $(DISTNAME)-$(DISTVERSION).zip HEAD
-
-pgxn-zip: $(DISTNAME)-$(DISTVERSION).zip
+pgxn-zip: dist.zip
```
Comment (ChuckHend) [inline Makefile:29]: Doesn't pgxn need this to be labeled with the version?

File: Trunk.toml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+documentation = "https://github.com/tembo-io/pgmq"
+categories = [
+      "mq",
+      "message queue",
+      "messaging",
+      "queues",
+      "fifo",
+      "sqs",
+      "rsmq"
+]
```
Comment (ChuckHend) [inline Trunk.toml:16]: @vrmiguel what are the possible enums for these categories? can we put any string that we want?

File: integration_test/src/lib.rs (hunk 2 above)
Comment (ChuckHend) [reply integration_test/src/lib.rs:1]: I think if we remove the `[lib]` section from Cargo.toml we can remove the `./src` dir and lib.rs.

Comment (v0idpwn) [reply integration_test/src/lib.rs]: Awesome!

File: Makefile (hunk 3 above)
Comment (v0idpwn) [reply Makefile]: This is just a reference to another target. I changed it from `$(DISTNAME)-$(DISTVERSION).zip` to simply `dist`. 

This was broken, though: it shouldn't be `dist.zip`, but `dist`. Good catch :)

File: Trunk.toml (hunk 4 above)
Comment (v0idpwn) [reply Trunk.toml:16]: Copied those from META.json

File: .github/workflows/extension_ci.yml [hunk 5]
```diff
@@ -39,72 +39,40 @@ jobs:
...
-          {version: "14.12", pgrx_version: "pg14"},
-          {version: "15.7", pgrx_version: "pg15"},
-          {version: "16.3", pgrx_version: "pg16"}
-        ]
+        pg: [12, 13, 14, 15, 16]
+    container: pgxn/pgxn-tools
     steps:
+      - run: pg-start ${{ matrix.pg }}
       - uses: actions/checkout@v2
+      - run: apt install -y pkg-config
```
Comment (theory) [inline .github/workflows/extension_ci.yml:58]: Hrm, what's this for? Maybe I should add it to pgxn-tools?

**Review by theory — APPROVED** [review]

File: Makefile [hunk 6]
```diff
@@ -1,27 +1,38 @@
-PGRX_POSTGRES ?= pg16
-DISTNAME = $(shell grep -m 1 '^name' Cargo.toml | sed -e 's/[^"]*"\([^"]*\)",\{0,1\}/\1/')
-DISTVERSION  = $(shell grep -m 1 '^version' Cargo.toml | sed -e 's/[^"]*"\([^"]*\)",\{0,1\}/\1/')
-PG_VERSION:=16
-PGRX_PG_CONFIG =$(shell cargo pgrx info pg-config ${PGRX_POSTGRES})
+EXTENSION    = pgmq
+EXTVERSION   = $(shell grep -oP "^default_version\s*=\s\K.*" pgmq.control | tr -d "'")
```
Comment (theory) [inline Makefile:2]: On macOS:

```
❯ make META.json
grep: invalid option -- P
usage: grep [-abcdDEFGHhIiJLlMmnOopqRSsUVvwXxZz] [-A num] [-B num] [-C[num]]
	[-e pattern] [-f file] [--binary-files=value] [--color=when]
	[--context[=num]] [--directories=action] [--label] [--line-buffered]
	[--null] [pattern] [file ...]
grep: invalid option -- P
usage: grep [-abcdDEFGHhIiJLlMmnOopqRSsUVvwXxZz] [-A num] [-B num] [-C[num]]
	[-e pattern] [-f file] [--binary-files=value] [--color=when]
	[--context[=num]] [--directories=action] [--label] [--line-buffered]
	[--null] [pattern] [file ...]
grep: invalid option -- P
usage: grep [-abcdDEFGHhIiJLlMmnOopqRSsUVvwXxZz] [-A num] [-B num] [-C[num]]
	[-e pattern] [-f file] [--binary-files=value] [--color=when]
	[--context[=num]] [--directories=action] [--label] [--line-buffered]
	[--null] [pattern] [file ...]
sed 's/@@VERSION@@//g' META.json.in > META.json
```

**Review by ChuckHend — APPROVED** [review]

File: Makefile [hunk 7]
```diff
@@ -1,27 +1,38 @@
...
 	docker run -d --name pgmq-pg -e POSTGRES_PASSWORD=postgres -p 5432:5432 quay.io/tembo/pgmq-pg:latest
 
-META.json: META.json.in Cargo.toml
-	@sed "s/@CARGO_VERSION@/$(DISTVERSION)/g" $< > $@
+pgxn-zip: dist
 
-$(DISTNAME)-$(DISTVERSION).zip: META.json
-	git archive --format zip --prefix $(DISTNAME)-$(DISTVERSION)/ --add-file $< -o $(DISTNAME)-$(DISTVERSION).zip HEAD
+META.json:
+	sed 's/@@VERSION@@/$(EXTVERSION)/g' META.json.in > META.json
```
Comment (theory) [inline Makefile:29]: This sets `$.provides.pgmq.version`, but not `$.version`.

**Review by theory — CHANGES_REQUESTED** [review]

File: .github/workflows/extension_ci.yml (hunk 5 above)
Comment (v0idpwn) [reply .github/workflows/extension_ci.yml:58]: Our tests are still written in rust, and needed this to compile.

File: Makefile (hunk 6 above)
Comment (v0idpwn) [reply Makefile:2]: Nice catch!

File: Makefile (hunk 7 above)
Comment (v0idpwn) [reply Makefile:29]: It replaces both. In my META.json template  I used `@@VERSION@@` for the two fields, as for us they will always be equal. I was also thinking of removing DISTVERSION and using EXTVERSION in everything, wdyt?

File: Trunk.toml (hunk 4 above)
Comment (vrmiguel) [reply Trunk.toml:16]: @ChuckHend the possible values are

```
            name             |           slug            
-----------------------------+---------------------------
 Tooling / Admin             | tooling_admin
 Metrics                     | metrics
 Change Data Capture         | change_data_capture
 Orchestration               | orchestration
 Procedural Languages        | procedural_languages
 Machine Learning            | machine_learning
 Analytics                   | analytics
 Connectors                  | connectors
 Data / Transformations      | data_transformations
 Index / Table Optimizations | index_table_optimizations
 Query Optimizations         | query_optimizations
 Auditing / Logging          | auditing_logging
 Debugging                   | debugging
 Featured                    | featured
 Search                      | search
 Security                    | security
```

Also available in https://registry.pgtrunk.io/categories/all

Comment (v0idpwn) [reply Trunk.toml:16]: Maybe "connectors"? But this seems to apply more to fdws... I don't think there's a single one that fully matches pgmq.

Comment (ChuckHend) [reply Trunk.toml:16]: It looks like we are currently using `["featured", "orchestration"]`.  https://pgt.dev/extensions/pgmq

Comment (ChuckHend) [reply Trunk.toml:16]: I gave wrong link. Here is where we are setting those categories
https://github.com/tembo-io/pgmq/blob/ced449fc6ac1bdf86d82c057653990de99fcd76b/.github/workflows/extension_ci.yml#L151

Comment (v0idpwn) [reply Trunk.toml:16]: Does --category override `Trunk.toml`? Should I maybe just remove the categories key from `Trunk.toml`?

File: Makefile (hunk 6 above)
Comment (v0idpwn) [reply Makefile]: (Apparently fixed, could you test again?)

File: images/pgmq-pg/Dockerfile.in (hunk 1 above)
Comment (ChuckHend) [reply images/pgmq-pg/Dockerfile.in:13]: Probably not...

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -30,56 +24,24 @@ git clone https://github.com/tembo-io/pgmq.git
...
 
 ```bash
-cargo pgrx run pg16
-```
-
-Finally, you can create the extension and get started with the example in the [README.md](README.md).
-
-```psql
-CREATE EXTENSION pgmq cascade;
+DB_URL=postgres:postgres@localhost:5432/postgres make test
```
Comment (ChuckHend) [inline CONTRIBUTING.md:44]:
```suggestion
DATABASE_URL=postgres:postgres@localhost:5432/postgres make test
```
